ed25519/signer_test.go

```diff
@@ -0,0 +1,295 @@
+package ed25519
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"testing"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_NewSigner(t *testing.T) {
+	t.Parallel()
+	const (
+		testKeyId      = "key-id"
+		testKeyPurpose = wrapping.KeyPurpose_Sign
+	)
+	testCtx := context.Background()
+	_, testPrivKey, err := ed25519.GenerateKey(rand.Reader)
+	require.NoError(t, err)
+	marshKey, err := x509.MarshalPKCS8PrivateKey(testPrivKey)
+	require.NoError(t, err)
+	tests := []struct {
+		name            string
+		opt             []wrapping.Option
+		wantSigner      *Signer
+		wantErr         bool
+		wantErrIs       error
+		wantErrContains string
+	}{
+		{
+			name: "success-with-opts",
+			opt: []wrapping.Option{
+				WithPrivKey(testPrivKey),
+				wrapping.WithKeyId(testKeyId),
+				wrapping.WithKeyPurposes(testKeyPurpose),
+				wrapping.WithKeyType(wrapping.KeyType_ED25519),
+			},
+			wantSigner: &Signer{
+				privKey:     testPrivKey,
+				keyPurposes: []wrapping.KeyPurpose{testKeyPurpose},
+				keyId:       testKeyId,
+				keyType:     wrapping.KeyType_ED25519,
+			},
+		},
+		{
+			name: "success-with-config",
+			opt: []wrapping.Option{
+				wrapping.WithConfigMap(map[string]string{
+					ConfigKeyId:       testKeyId,
+					ConfigKeyPurposes: wrapping.KeyPurpose_name[int32(wrapping.KeyPurpose_Sign)],
+					ConfigPrivKey:     string(pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: marshKey})),
+				}),
+			},
+			wantSigner: &Signer{
+				privKey:     testPrivKey,
+				keyPurposes: []wrapping.KeyPurpose{testKeyPurpose},
+				keyId:       testKeyId,
+				keyType:     wrapping.KeyType_ED25519,
+			},
+		},
```

Comment:
Could you add a test to assert that local options override config map options? E.g. provide the private key in both and ensure the local one prevails.